src/ASM/ASMs2D.C

```diff
@@ -3068,3 +3068,25 @@ size_t ASMs2D::getNoProjectionNodes () const
 {
   return proj->numCoefs_u() * proj->numCoefs_v();
 }
+
+
+void ASMs2D::getNeighbours (NeighArray& neigh) const
+{
+  int N1, N2, dummy;
+  this->getNoStructElms(N1, N2, dummy);
+  for (int i = 0; i < static_cast<int>(this->getNoElms(true)); ++i) {
+    int idx = this->getElmID(i+1)-1;
+    if (idx == -1)
+      continue;
+
+    neigh[idx].resize(4,-1);
```

Comment:
So the big assumption here is that `neigh` has been resized outside to big enough?

Reply:
yes. this is called from the SIM and the SIM call make sure of that.